mongodb/Acornfile

```diff
@@ -0,0 +1,585 @@
+args: {
+    // Specify whether MongoDB instance is a standalone or replica set
+    isReplicaSet: false
+    // Custom user to be created during the initialization
+    dbUserName: ""
+    // Custom database to be created during the initialization
+    dbName: "acorn"
+    // Number of secondary nodes
+    replicas: 3
+    // Enable diagnostic mode
+    diagnosticMode: false
+    // Enable authentication
+    authEnabled: true
+    // Mongodb additional command line flags
+    extraFlags: ""
+    // Run arbiter node
+    arbiterEnabled: false
+    // Arbiter additional command line flags
+    arbiterExtraFlags: ""
+    // Number of hidden nodes
+    hiddenReplicas: 0
+    // Hidden node additional command line flags
+    hiddenExtraFlargs: ""
+    // App installed namespace. NOTE(joseb): before fieldRef is supported,
+    appNamespace: "mongo"
+    // Backup Schedule
+    backupSchedule: ""
+    // Specify to use oplog for taking a point-in-time snapshot
+    ptrBackup: false
+    // Collection name to backup
+    backupCollection: ""
+    // DB name to backup
+    backupDB: ""
+    // Backup name to restore
+    backupToRestore: ""
+    // Mongodb version
+    dbVersion: "5.0.12"
+    // Days to keep backup
+    backupRetainDays: 5
+    // Allow access from outside of k8s cluster
+    externalAccessEnabled: false
+    // TLS certificate

```

Comment:
The cert and keys should be bound in as secrets. This is sensitive info. The mode is a good option to have though.

Reply:
@cloudnautique 
I checked this already https://docs.acorn.io/running/certificates#consume-the-secret, 
But I think this is just for http endpoints and also offloads TLS at acorn or ingress level, not app level.
So, we cannot use that way. 

https://github.com/acorn-io/library/pull/22/files#diff-35b4d1b980762a71704bfd1b7863ca8d9c1888fbc3db8be708e907d070da5792R577-R583 
ofc, I created a secret using tls key and certs, and used that secret in container and jobs. So, you mean we have to just keep one Arg of pre-existing k8s secret name (for example, `tlsSecretName`) instead of Args of bare key and certs?
In this case, I wonder if pre-existing k8s secrets can be referred in Acornfile.

Reply:
Yeah if there is a secret named `foo` in the Acornfile that is expected to be a cert when the user runs the app they can bind it in.. `acorn run -s my-existing-cert-secret:foo [MONGOIMAGE]`. https://docs.acorn.io/running/args-and-secrets#binding-a-secret-at-runtime

Just let Acorn handle it, no need to make an argument for the user to pass in.

Reply:
aha cool
So in this case, just need to create a empty secret like this?
```
    if localData.tlsEnabled {
        "tls-certs": {
            type: "opaque"
        }
    }
```
Or at least need to specify the keys in the secret?
```
    if localData.tlsEnabled {
        "tls-certs": {
            type: "opaque"
            data: {
                "cert-key": ""
                "ca": ""
            }
        }
    }
```